src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -527,19 +527,28 @@ class DatatypeSegment(BaseSegment):
         Sequence(
             OneOf("time", "timestamp"),
             Bracketed(Ref("NumericLiteralSegment"), optional=True),
-            Sequence(OneOf("WITH", "WITHOUT"), "TIME", "ZONE", optional=True),
+            OneOf(
+                Sequence(OneOf("WITH", "WITHOUT"), "TIME", "ZONE"),
+                optional=True,
+            ),
```

Comment:
Might as well revert this change now?

```suggestion
            Sequence(OneOf("WITH", "WITHOUT"), "TIME", "ZONE"), optional=True),
```